packages/nextjs/components/scaffold-eth/Input/AddressInput.tsx

```diff
@@ -51,34 +62,74 @@ export const AddressInput = ({ value, name, placeholder, onChange, disabled }: C
     [onChange],
   );
 
+  const addressBook = useAddressBook();
```

Comment:
I don't want to run this hooks logic always, we can maybe pass `showAddressBook` to this hook and make RPC calls only if `showAddressBook` is true since as discussed there can be a lot of public addresses and we should avoid unnecessary RPC calls 🙌